pkgs/development/python-modules/jaxlib/default.nix

```diff
@@ -236,13 +238,25 @@ let
         for src in ./jaxlib/*.{cc,h}; do
           sed -i 's@include/pybind11@pybind11@g' $src
         done
-        sed -i 's@-lprotobuf@-l:libprotobuf.a@' ../output/external/org_tensorflow/third_party/systemlibs/protobuf.BUILD
-        sed -i 's@-lprotoc@-l:libprotoc.a@' ../output/external/org_tensorflow/third_party/systemlibs/protobuf.BUILD
         substituteInPlace ../output/external/org_tensorflow/tensorflow/compiler/xla/python/pprof_profile_builder.cc \
           --replace "status.message()" "std::string{status.message()}"
       '' + lib.optionalString cudaSupport ''
         patchShebangs ../output/external/org_tensorflow/third_party/gpus/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc.tpl
-      '';
+      '' + lib.optionalString stdenv.isDarwin ''
+        # Framework search paths aren't added by bintools hook
+        # https://github.com/NixOS/nixpkgs/pull/41914
+        export NIX_LDFLAGS+=" -F${IOKit}/Library/Frameworks"
+        substituteInPlace ../output/external/rules_cc/cc/private/toolchain/unix_cc_configure.bzl \
+          --replace "/usr/bin/libtool" "${cctools}/bin/libtool"
+        substituteInPlace ../output/external/rules_cc/cc/private/toolchain/osx_cc_wrapper.sh.tpl \
+          --replace "/usr/bin/install_name_tool" "${cctools}/bin/install_name_tool"
+      '' + (if stdenv.cc.isGNU then ''
+        sed -i 's@-lprotobuf@-l:libprotobuf.a@' ../output/external/org_tensorflow/third_party/systemlibs/protobuf.BUILD
+        sed -i 's@-lprotoc@-l:libprotoc.a@' ../output/external/org_tensorflow/third_party/systemlibs/protobuf.BUILD
+      '' else if stdenv.cc.isClang then ''
+        sed -i 's@-lprotobuf@${pkgs.protobuf}/lib/libprotobuf.a@' ../output/external/org_tensorflow/third_party/systemlibs/protobuf.BUILD
+        sed -i 's@-lprotoc@${pkgs.protobuf}/lib/libprotoc.a@' ../output/external/org_tensorflow/third_party/systemlibs/protobuf.BUILD
```

Comment:
Why do we link against `pkgs.protobuf` when using clang, but the vendored version when building with gcc?

Reply:
Both are using `pkgs.protobuf`, we pass `com_google_protobuf` in `TF_SYSTEM_LIBS` to configure the build to use protobuf libraries and headers from the system (https://github.com/tensorflow/tensorflow/blob/144d3dc0929569224728c7634ccd93c8a1bd3da9/third_party/repo.bzl#L66), by stubbing out the vendored build targets with targets that only set compiler and linker flags (https://github.com/tensorflow/tensorflow/blob/144d3dc0929569224728c7634ccd93c8a1bd3da9/third_party/systemlibs/protobuf.BUILD#L62-L78).

It's true that by hardcoding the path here it makes it impossible for hooks to affect it, compared to what we do for gcc, but hopefully it's ok.

Reply:
This is subtle enough that I think it may deserve a comment, but I don't think it's necessarily a blocker for getting this merged.

Reply:
Updated the existing comment to clarify this bit.